src/ffi.rs

```diff
@@ -177,8 +179,9 @@ pub unsafe extern "C" fn context_add_value(
 
     let value: Result<Value, _> = value.try_into();
     if let Err(e) = value {
-        errbuf[..e.len()].copy_from_slice(e.as_bytes());
-        *errbuf_len = e.len();
+        let errlen = min(e.len(), ERR_BUF_MAX_LEN);
```

Comment:
We will probably want a similar test case for `context:add_value()` to cover this.

Reply:
I double checked and the only error that could happen with `add_value` is `Utf8Error`, which can not possibly exceed `2048` bytes due to the formatting:

https://doc.rust-lang.org/src/core/str/error.rs.html#115-122

So maybe we only need to handle the parsing error. You should not even be able write a test case for `add_value` that triggers this bug.